plugins/experimental/rate_limit/sni_selector.cc

```diff
@@ -16,117 +16,240 @@
  * limitations under the License.
  */
 #include "tscore/ink_config.h"
+#include <yaml-cpp/yaml.h>
 
-#include <cstring>
-
-#include "sni_limiter.h"
 #include "sni_selector.h"
 
+std::atomic<SniSelector *> SniSelector::_instance = nullptr;
+
 ///////////////////////////////////////////////////////////////////////////////
-// This is the queue management continuation, which gets called periodically
+// YAML parser for the global YAML configuration (via plugin.config)
 //
-static int
-sni_queue_cont(TSCont cont, TSEvent event, void *edata)
+bool
+SniSelector::yamlParser(const std::string &yaml_file)
 {
-  SniSelector *selector = static_cast<SniSelector *>(TSContDataGet(cont));
+  YAML::Node config;
+
+  try {
+    config = YAML::LoadFile(yaml_file);
+  } catch (YAML::BadFile &e) {
```

Comment:
You are not going to change the exception right? if not then why not `const`.
the one below too.

Reply:
I just copied this pattern from other places in the code :). I can change it.

Reply:
hope it wasn't my code! that would be a shame :rofl: